core/server/api/canary/redirects.js

```diff
@@ -25,7 +25,11 @@ module.exports = {
         },
         permissions: true,
         response: {
-            format: 'plain'
+            format() {
+                const filePath = redirects.settings.getCurrentRedirectsFilePathSync();
```

Comment:
Hm... not really convinced about having `sync` method here... I know it's necessary in this situation, but would block the even loop. There could be couple solutions here:
1. Add support for async `response.format()` handling, then we'd be able to return an promise here and use async method to figure out redirects file path.
2. "cache" file path inside of `redirects.settings` module and return cached value here

Reply:
I tried the cache method. But I decided not to use it because users can change json to yaml or vice versa with other tools like ftp, telnet. And it makes testing almost impossible. 

That's why I implemented the 1.

Reply:
If someone is changing the file directly we usually assume the user knows what they are doing and would restart the instance manually. For example, if you modify `routes.yaml` file right now through ftp these changes won't kick in till restart. Cache should be refreshed when the user uploads a new file through API, that's the only way we can control it and should be enough.

Reply:
For now, I think the async method will be enough and we can think about caching the response if it becomes problematic. Sites rarely need to change these redirects after initial setup.